router/router.go

```diff
@@ -0,0 +1,220 @@
+package router
+
+import (
+	"encoding/json"
+	"errors"
+	"iotdashboard/controller"
+	"log"
+	"net"
+	"net/http"
+)
+
+type RouterService struct {
+	Ctrlr     *controller.ControllerService
+	httpPort  string
+	httpsPort string
+	certPath  string
+	keyPath   string
+}
+
+// Credentials is a struct that holds the email, password, and CSRF token of a request
+type Credentials struct {
+	Email    string `json:"email"`
+	Password string `json:"password"`
+	CSRF     string `json:"csrf"`
+}
+
+func NewRouter(httpPort, httpsPort, certPath, keyPath string) (*RouterService, error) {
+	Ctrlr, err := controller.NewController()
+	if err != nil {
+		return &RouterService{}, err
+	}
+	return &RouterService{Ctrlr, httpPort, httpsPort, certPath, keyPath}, nil
+}
+
+func (rtr *RouterService) Start() error {
+	log.Printf("Starting webserver ... \n")
+	//start listening for http to redirect to https
+	go log.Print(http.ListenAndServe(rtr.httpPort, http.HandlerFunc(rtr.redirectTLS)))
```

Comment:
Without a wrapper func, this line would block until `http.ListenAndServe` returns.
When you spawn a goroutine, it evaluates all the arguments synchronously (the return error from `http.ListenAndServe` here for example) before kicking off the goroutine and proceeding.
```suggestion
	go func() {
		log.Print(http.ListenAndServe(rtr.httpPort, http.HandlerFunc(rtr.redirectTLS)))
	}()
```